injector/src/main/java/edu/ucr/cs/riple/injector/location/OnMethod.java

```diff
@@ -41,12 +42,16 @@ public class OnMethod extends Location {
   public final String method;
   public final SignatureMatcher matcher;
 
-  public OnMethod(String path, String clazz, String method) {
+  public OnMethod(Path path, String clazz, String method) {
     super(LocationType.METHOD, path, clazz);
     this.method = method;
     this.matcher = new SignatureMatcher(method);
   }
 
+  public OnMethod(String path, String clazz, String method) {
+    this(Helper.deserializePath(path), clazz, method);
```

Comment:
Why have both this and the explicit call to `deserializePath` in `MethodDeclarationTree.java`? Maybe is better to only take a `Path` in the constructor here and force the caller to deserialize?

Reply:
Actually I wanted to that, but then we have to change expected output for all unit tests of `Injector` (more than 100 tests). I thought this might make this PR too large and we can do it in follow up. Please let me know if you think it is better to do it in the same PR. Thank you.